### PR TITLE
Add git info of exadg

### PR DESCRIPTION
**Pull request by jh66637**
I am currently trying to trace back the commit where I seem to have introduced a bug in a related project. Having the info of the branch on which results were computed would be very helpful. This PR adds the info to the output.

### CODE REVIEW

File: cmake/macros/macro_git_info.cmake
Comment (peterrum) [inline cmake/macros/macro_git_info.cmake:28]: Just use: https://github.com/hyperdeal/hyperdeal/blob/8145d0a3e90aa479d0ea158991caf3886ec17dd1/CMakeLists.txt#L61

File: include/exadg/utilities/exadg_git_info.cpp
Comment (peterrum) [inline include/exadg/utilities/exadg_git_info.cpp:25]: Could we keep the interface similar is it in deal.II: https://github.com/dealii/dealii/blob/master/include/deal.II/base/revision.h.in?

**Review by peterrum — COMMENTED** [review]
Revision :1st_place_medal:

File: include/exadg/utilities/exadg_git_info.cpp
Comment (jh66637) [reply include/exadg/utilities/exadg_git_info.cpp:25]: @peterrum I can do that, but I would like to keep the additional ```.cpp``` file. Otherwise, after every commit, there might be a lot of headers that have to be rebuilt. I can unify this, such that also the header of the ```deal.II``` revision is included in the source file.
Any comments? :)

Comment (peterrum) [reply include/exadg/utilities/exadg_git_info.cpp:25]: Sure :+1:

File: include/exadg/utilities/print_general_infos.h
Comment (nfehn) [inline include/exadg/utilities/print_general_infos.h:104]: We should discuss why this is done differently than for deal.II a few lines above. It is not clear to me why we introduce `exadg_git_info.h`, `exadg_git_info.cpp` in this PR? One solution could be to change it on the deal.II side (with changes in the deal.II library itself and/or get-functions in ExaDG).

Comment (jh66637) [reply include/exadg/utilities/print_general_infos.h:104]: I would like to include the file which contains the #defines (currently strings, but I will change it) in one source file only, to avoid recompilation of several files if the commit hash changes. Therefore, I introduced the get functions. It would be good to encapsulate the files for deal.ii as well, but it is not as important since the deal.ii library is probably less often recompiled if the focus is on developing code for ExaDG. Personally, I don't think this needs a change on the deal.ii side. 

I will update this PR as soon as possible such that it could be merged

**Comment by jh66637** [comment]
@peterrum I updated such that ```deal.ii``` and ```ExaDG``` have the same interface. @nfehn To keep the revision header includes in a source file only I added ```print_general_infos.cpp``` in which I put all functions that are no templates. 

From my side, this would be ready to merge once all tests have passed. What do you think?

File: include/exadg/utilities/print_general_infos.cpp
Comment (nfehn) [inline include/exadg/utilities/print_general_infos.cpp:78]: what about this comment here?

File: include/exadg/utilities/print_general_infos.h
Comment (nfehn) [inline include/exadg/utilities/print_general_infos.h:29]: Why including mapping here? Is it really needed by this file or somewhere else?

File: include/exadg/utilities/print_general_infos.h
Comment (nfehn) [inline include/exadg/utilities/print_general_infos.h:119]: I would not print ExaDG info between deal.II and matrix-free info. Maybe we print ExaDG first, given that its the ExaDG software here and given that deal.II/matrix-free is additional third-party information.

File: include/exadg/utilities/print_general_infos.h
Comment (jh66637) [reply include/exadg/utilities/print_general_infos.h:29]: It is needed in ```print_grid_info()``` and after splitting the files, there were errors that MappingQ is not known

File: include/exadg/utilities/print_general_infos.cpp
Comment (jh66637) [reply include/exadg/utilities/print_general_infos.cpp]: I resolved this

File: include/exadg/utilities/print_general_infos.h
Comment (jh66637) [reply include/exadg/utilities/print_general_infos.h:118]: Done :)

File: include/exadg/utilities/print_general_infos.h
Comment (nfehn) [reply include/exadg/utilities/print_general_infos.h:29]: I see, we should in the future shift the other implementations also to the cpp file.

**Review by nfehn — APPROVED** [review]

**Comment by nfehn** [comment]
Let's wait for @peterrum 's approval.

**Review by peterrum — APPROVED** [review]

**Pull request merged**
